crates/math/src/rows_batch.rs

```diff
@@ -35,30 +37,77 @@ impl<'a, T> RowsBatch<'a, T> {
 		RowsBatchRef {
 			rows: self.rows.as_slice(),
 			row_len: self.row_len,
+			offset: 0,
 		}
 	}
 }
 
-/// This struct is similar to `RowsBatch`, but it holds a reference to the slice of rows.
+/// This struct is similar to `RowsBatch`, but it holds a reference to the slice of rows
+/// and an offset.
+///
+/// It is guaranteed that all rows have the length at least `row_len + offset`
+///
 /// Unfortunately due to liftime issues we can't have a single generic struct which is
```

Comment:
typo: liftime -> lifetime

Also, please change this comment to clearly explain the connection to `RowsBatch`. Now that there's a documentation line between the first and last lines, it's confusing what this comment refers to.